pkg/local_object_storage/engine/delete.go

```diff
@@ -102,5 +111,46 @@ func (e *StorageEngine) delete(prm DeletePrm) (DeleteRes, error) {
 		return DeleteRes{}, locked.err
 	}
 
+	if splitInfo != nil {
+		e.deleteChildren(prm.addr, prm.forceRemoval, splitInfo.SplitID())
+	}
+
 	return DeleteRes{}, nil
 }
+
+func (e *StorageEngine) deleteChildren(addr oid.Address, force bool, splitID *objectSDK.SplitID) {
+	var fs objectSDK.SearchFilters
+	fs.AddSplitIDFilter(objectSDK.MatchStringEqual, splitID)
+
+	var shPrm shard.InhumePrm
+	if force {
+		shPrm.ForceRemoval()
+	}
+
+	e.iterateOverSortedShards(addr, func(_ int, sh hashedShard) (stop bool) {
+		var prm shard.SelectPrm
```

Comment:
Maybe init it once outside same as `shPrm`?

Reply:
FIxed

Reply:
`addr` seems static, too.

Reply:
Fixed